.github/workflows/_linux_ut.yml

```
@@ -51,17 +51,158 @@ jobs:
+              for test in $(ls test/xpu | grep test); do test_cmd="${test_cmd} xpu/$test"; done
+              if [ -f "test/test_xpu.py" ]; then test_cmd="${test_cmd} test_xpu.py"; fi
+              eval $test_cmd
+            log_prefix: 'torch_xpu'
+            timeout: 10000
+            env:
+              PYTORCH_TEST_WITH_SLOW: '1'
+              PYTORCH_TESTING_DEVICE_ONLY_FOR: 'xpu'
+            additional_steps: |
+              TEST_REPORTS_DIR=${{ github.workspace }}/../pytorch/test/test-reports
+              rm -rf "$TEST_REPORTS_DIR" && mkdir -p "$TEST_REPORTS_DIR"
+          - name: 'xpu_profiling'
+            condition: ${{ inputs.driver == 'rolling' && contains(inputs.ut, 'xpu_profiling') }}
+            directory: '${{ github.workspace }}'
+            command_script: |
+              # RN50 Test
+              PROFILE=1 python -u test/profiling/rn50.py -a resnet50 --dummy ./ --num-iterations 20 --xpu 0
+              cp profiling.fp32.train.pt $LOG_DIR/profile_test
+
+              # All Issue Reproduce UT
+              python -u test/profiling/correlation_id_mixed.py | \
+                tee $LOG_DIR/profile_test/issue_reproduce/correlation_id_mixed.log
+              python -u test/profiling/reproducer.missing.gpu.kernel.time.py | \
+                tee $LOG_DIR/profile_test/issue_reproduce/reproducer.missing.gpu.kernel.time.log
+              python -u test/profiling/time_precision_in_profile.py | \
+                tee $LOG_DIR/profile_test/issue_reproduce/time_precision_in_profile.log
+              python -u test/profiling/profile_partial_runtime_ops.py | \
+                tee $LOG_DIR/profile_test/issue_reproduce/profile_partial_runtime_ops.log
+              python -u test/profiling/triton_xpu_ops_time.py | \
+                tee $LOG_DIR/profile_test/issue_reproduce/triton_xpu_ops_time.log
+
+              # All xpu ut under test/profiler
+              cd ../pytorch/test/profiler
+              python -m pytest --timeout 600 -vs test_cpp_thread.py | \
+                tee $LOG_DIR/profile_test/test_cpp_thread.log
+              python -m pytest --timeout 600 -vs test_execution_trace.py | \
+                tee $LOG_DIR/profile_test/test_execution_trace.log
+              python -m pytest --timeout 600 -vs test_memory_profiler.py | \
+                tee $LOG_DIR/profile_test/test_memory_profiler.log
+              python -m pytest --timeout 600 -vs test_profiler_tree.py | \
+                tee $LOG_DIR/profile_test/test_profiler_tree.log
+            timeout: 10000
+            env: {}
+            additional_steps: |
+              mkdir -p $LOG_DIR/profile_test/issue_reproduce
     steps:
       - name: Checkout torch-xpu-ops
         uses: actions/checkout@v4
+      - name: Create unique workspace
+        run: |
+          # Create unique conda env for each UT test
+          echo "CONDA_ENV_NAME=xpu_op_${ZE_AFFINITY_MASK}_${{ matrix.test.name }}" >> $GITHUB_ENV
```

That's not truly unique actually. Meaning that if there are 2 of these workloads running in parallel, for example for different PRs, these will start to fail. To address this you can reuse approach from #1282. It will work only with Linux though. For Windows you will need something else:
```suggestion
          random=$(head /dev/urandom | tr -dc A-Za-z0-9_ | head -c ${1:-5} | xargs)
          echo "CONDA_ENV_NAME=xpu_op_${ZE_AFFINITY_MASK}_${{ matrix.test.name }}_${random}" >> $GITHUB_ENV
```

---

Got it. Thank you so much for your help~~